### DISCUSSION TITLE
A question about  pull request #561: [Support re-transform/hot-swap classes with other Java agents].

### DISCUSSION THREAD

**Discussion by jiangyunpeng** [apache/skywalking#11857] [Q&A]
We have been using Skywalking 8.x for some time, and we are now considering upgrading to version 9.x. While investigating this upgrade, we came across an interesting feature mentioned in pull request [#561](https://github.com/apache/skywalking-java/pull/561): [Support re-transform/hot-swap classes with other Java agents].

We are curious about the differences between cache_enhanced_class and this pull request. Has this pull request resolved the issue of Arthas being unable to redefine (hot deploy) Spring MVC controllers with Skywalking? And are there any issues link discuss about this pull request ? 

Thanks!

**Comment by wu-sheng** [comment]
@kylixs should be able to help here.

I would say, 9.x increases the compatibility with Arthas, but not sure about hot load.
